corehq/apps/sso/forms.py

```diff
@@ -631,6 +707,34 @@ def clean_entity_id(self):
         _ensure_entity_id_matches_expected_provider(entity_id, self.idp)
         return entity_id
 
+    def clean_api_secret(self):
+        api_secret = self.cleaned_data['api_secret']
```

Comment:
@mjriley this is standard procedure in django forms ([see docs](https://docs.djangoproject.com/en/5.0/ref/forms/validation/#cleaning-a-specific-field-attribute))

Reply:
@jingcheng16 I do have concerns about the implementation here, as it seems the actual api secret is still being passed into the javascript. While the user can't see it, it is in plaintext on the page.

Here's how I thought validation would go for `api_secret`:
If `enable_user_deactivation` is `True` the return a `ValidationError` if `api_secret` is `''` string and `self.idp.api_secret` is `''`

We never pass the actual `api_secret` to the javascript. Instead, when the page is initialized, we pass a boolean to the `ko` (eg `show_update_api_secret`) if we detected that `self.idp.api_secret` was `''` when initializing the form.

When saving the form data to the `IdentityProvider` we ONLY update the value of `api_secret` if the form presents a non-empty value. Otherwise, if the value of `self.cleaned_data['api_secret']` is `''`, do not update the value that is set for `self.idp.api_secret`